goal_src/jak2/engine/camera/cam-debug.gc

```diff
@@ -5,5 +5,2137 @@
+      (format arg1 "BUTT_CAM ")
+      )
+  (if (= (logand (cam-slave-options VERTICAL_FOLLOW_MATCHES_CAMERA) arg0)
+         (cam-slave-options VERTICAL_FOLLOW_MATCHES_CAMERA)
+         )
+      (format arg1 "VERTICAL_FOLLOW_MATCHES_CAMERA ")
+      )
+  (if (= (logand (cam-slave-options HAVE_BUTT_HANDLE) arg0) (cam-slave-options HAVE_BUTT_HANDLE))
+      (format arg1 "HAVE_BUTT_HANDLE ")
+      )
+  (the-as string arg1)
+  )
+
+;; definition for function cam-index-options->string
+;; INFO: Return type mismatch object vs string.
+(defun cam-index-options->string ((arg0 cam-index-options) (arg1 object))
+  (if (= (logand arg0 (cam-index-options SPHERICAL)) (cam-index-options SPHERICAL))
+      (format arg1 "RADIAL ")
+      )
+  (if (= (logand arg0 (cam-index-options RADIAL)) (cam-index-options RADIAL))
+      (format arg1 "SPHERICAL ")
+      )
+  (the-as string arg1)
+  )
+
+;; definition for function slave-los-state->string
+(defun slave-los-state->string ((arg0 slave-los-state))
+  (case arg0
+    (((slave-los-state between))
+     "between"
+     )
+    (((slave-los-state none))
+     "none"
+     )
+    (((slave-los-state cw))
+     "cw"
+     )
+    (((slave-los-state ccw))
+     "ccw"
+     )
+    (else
+      "*unknown*"
+      )
+    )
+  )
+
+;; definition for function cam-line-dma
+;; INFO: Used lq/sq
+(defun cam-line-dma ()
+  (let* ((v1-5 (-> *display* frames (-> *display* on-screen) debug-buf))
+         (a2-0 (-> v1-5 base))
+         )
+    (let ((a0-1 (the-as object (-> v1-5 base))))
+      (let* ((a1-0 v1-5)
+             (a3-0 (the-as object (-> a1-0 base)))
+             )
+        (set! (-> (the-as dma-packet a3-0) dma) (new 'static 'dma-tag :id (dma-tag-id cnt)))
+        (set! (-> (the-as dma-packet a3-0) vif0) (new 'static 'vif-tag))
+        (set! (-> (the-as dma-packet a3-0) vif1) (new 'static 'vif-tag :cmd (vif-cmd direct) :msk #x1))
+        (set! (-> a1-0 base) (&+ (the-as pointer a3-0) 16))
+        )
+      (let* ((a1-1 v1-5)
+             (a3-2 (the-as object (-> a1-1 base)))
+             )
+        (set! (-> (the-as gs-gif-tag a3-2) tag)
+              (new 'static 'gif-tag64
+                :nloop #x1
+                :eop #x1
+                :pre #x1
+                :prim (new 'static 'gs-prim :prim (gs-prim-type line) :iip #x1 :abe #x1)
+                :nreg #x4
+                )
+              )
+        (set! (-> (the-as gs-gif-tag a3-2) regs)
+              (new 'static 'gif-tag-regs
+                :regs0 (gif-reg-id rgbaq)
+                :regs1 (gif-reg-id xyzf2)
+                :regs2 (gif-reg-id rgbaq)
+                :regs3 (gif-reg-id xyzf2)
+                )
+              )
+        (set! (-> a1-1 base) (&+ (the-as pointer a3-2) 16))
+        )
+      (let* ((a1-2 v1-5)
+             (a3-4 (-> a1-2 base))
+             )
+        (set! (-> (the-as (pointer uint128) a3-4) 0) (-> (the-as vector (+ 32 #x70000000)) quad))
```

Comment:
this will need manual fix up for scratchpad stuff, but we can can put it off for now